app/src/main/java/com/example/rickandmortyapp/view/MainActivity.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.rickandmortyapp.view
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.rickandmortyapp.viewmodel.CharacterAdapter
+import com.example.rickandmortyapp.databinding.ActivityMainBinding
+import com.example.rickandmortyapp.viewmodel.FetchCharacters
+
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+    private val characterAdapter: CharacterAdapter by lazy {
+        CharacterAdapter()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        FetchCharacters.fetchCharacters{ characterAdapter.submitList(it) }
```

Comment:
later we can thinkg about creating a cache to no call all the time the API. Also, starting to use flow/livedata for this.